runtime/jit_vm/cthelpers.cpp

```diff
@@ -78,6 +78,48 @@ jitGetInterfaceITableIndexFromCP(J9VMThread *currentThread, J9ConstantPool *cons
 	return interfaceClass;
 }
 
+/* Only returns non-null if the method is not to be dispatched by itable, i.e.
+ * if it is:
+ * - private, using direct dispatch;
+ * - a final method of Object, using direct dispatch; or
+ * - a non-final method of Object, using virtual dispatch.
+ */
+J9Method*
+jitGetImproperInterfaceMethodFromCP(J9VMThread *currentThread, J9ConstantPool *constantPool, UDATA cpIndex)
+	{
+	J9RAMInterfaceMethodRef *ramMethodRef = (J9RAMInterfaceMethodRef*)constantPool + cpIndex;
+	/* interfaceClass is used to indicate 'resolved'. Make sure to read it first */
+	J9Class* volatile interfaceClass = (J9Class*)ramMethodRef->interfaceClass;
+	VM_AtomicSupport::readBarrier();
+	UDATA volatile methodIndexAndArgCount = ramMethodRef->methodIndexAndArgCount;
+	J9Method *improperMethod = NULL;
+	/* If the ref is resolved, do not call the resolve helper, as it will currently fail if the interface class is not initialized */
+	if (NULL == interfaceClass) {
+		J9RAMInterfaceMethodRef localEntry;
+		if (0 == currentThread->javaVM->internalVMFunctions->resolveInterfaceMethodRefInto(currentThread, constantPool, cpIndex, J9_RESOLVE_FLAG_JIT_COMPILE_TIME, &localEntry)) {
```

Comment:
`resolveInterfaceMethodRefInto()` returns NULL or a `J9Method*`.  Please use NULL rather than 0 in the if condition

Reply:
This is a copy of already bad code, so we should fix both or neither.

Reply:
+1 to both =)

Reply:
Fixed (here and in `jitGetInterfaceITableIndexFromCP`)